### PR TITLE
[Arista] Add new hwskus to x86_64-arista_7060dx5_32

### PR DESCRIPTION
**Pull request by byu343**
#### Why I did it
Add two new hwskus for different port speed layouts
 - Arista-7060DX5-32-25Gx96-100Gx8-200Gx8
 - Arista-7060DX5-32-200Gx50-100Gx14

Disable bfd on all hwskus for x86_64-arista_7060dx5_32 as its dependencies have not been ready, which will result in a runtime error if not disabled.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How to verify it
Tested that interfaces are up on the two new HwSkus

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [x] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-200Gx50-100Gx14/hwsku.json [hunk 1]
```diff
@@ -0,0 +1,104 @@
+{
```
Comment (zhenggen-xu) [inline device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-200Gx50-100Gx14/hwsku.json:1]: This should be "200Gx48-100Gx32", i,e, port 1-24 has 2x200G and port 25-32 have 4x100G. please change the file accordingly.

File: device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-200Gx50-100Gx14/th4-a7060dx5-32.config.bcm [hunk 2]
```diff
@@ -0,0 +1,813 @@
+# configuration yaml file
```
Comment (zhenggen-xu) [inline device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-200Gx50-100Gx14/th4-a7060dx5-32.config.bcm:1]: Is this file shared for all hwskus?   If so, please move it to platform directory and let the sai.profile to reference to the same file.  If not I would recommend to rename the file to different names according to hwskus.

File: device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini [hunk 3]
```diff
@@ -0,0 +1,114 @@
+#name          lanes              alias        index  speed     fec
```
Comment (zhenggen-xu) [inline device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini:1]: Is this file required?

File: device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+Ethernet0      1                  Ethernet1/1  1      25000
```
Comment (zhenggen-xu) [inline device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini:2]: we should make 25G fec with fc,  100G  and 200G with rs fec.

**Review by zhenggen-xu — CHANGES_REQUESTED** [review]

File: device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-200Gx50-100Gx14/hwsku.json (hunk 1 above)
Comment (byu343) [reply device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-200Gx50-100Gx14/hwsku.json]: Has been changed to 200Gx48-100Gx32. We haven't fully tested the 100G ports after the change for no physical 100G-2 XCVR on the DUT, so far it is only verified that there is no error after the port initialization.

File: device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-200Gx50-100Gx14/th4-a7060dx5-32.config.bcm (hunk 2 above)
Comment (byu343) [reply device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-200Gx50-100Gx14/th4-a7060dx5-32.config.bcm]: It is not the same for all hwskus, so the filenames have been changed to be different.

File: device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini (hunk 4 above)
Comment (byu343) [reply device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini:2]: For 25G, FC looks not supported by the switch chip from the return of SAI_PORT_ATTR_SUPPORTED_FEC_MODE (NONE and RS supported). We will further check if it is hardware constraint or a software issue that we can fix. We did not specify it here to use the default value from switch chip (which is NONE for 25G) . User can configure/override the FEC config by /etc/sonic/config_db.json, as long as it is supported.

File: device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini (hunk 3 above)
Comment (zhenggen-xu) [reply device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini:1]: which lane pattern is not supported?  4x25G?  would be good to know if there is a limitation in the design?

File: device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini (hunk 4 above)
Comment (zhenggen-xu) [reply device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini:2]: thanks, can you please check with the vendor?  I am concerned even if we do it later by configDB, it will hit the same issue.

Comment (byu343) [reply device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini:2]: It is confirmed that FC FEC is not supported by TH4 chip.

File: device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini (hunk 3 above)
Comment (byu343) [reply device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini:1]: The 4 sub-ports of the first FP port are using lane 1, 5, 3, 7
#name          lanes              alias        index  speed     fec
Ethernet0      1                  Ethernet1/1  1      25000
Ethernet1      5                  Ethernet1/2  1      25000
Ethernet2      3                  Ethernet1/3  1      25000
Ethernet3      7                  Ethernet1/4  1      25000

In platform.json, the first FP port is defined with lane 1, 2, 3, 4, 5, 6, 7, 8: https://github.com/sonic-net/sonic-buildimage/blob/master/device/arista/x86_64-arista_7060dx5_32/platform.json#L221

For the syntax of breakout_mode, we cannot define a new breakout_mode in platform.json to represent the pattern of 1, 5, 3, 7 for the 4 sub-ports

Comment (zhenggen-xu) [reply device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini:1]: Hmm,  the use case is to use QSFP DAC cable (not QSFPDD),  and the physical breakout port 1 map to Ethernet0, breakout port 2 map to Ethernet1 and so on.  Can you confirm if that is the mapping in your design?

Comment (byu343) [reply device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini:1]: Yes, physical port 1, 2, 3, 4 on the breakout cable are mapped to Ethernet0, 1, 2, 3 in SONiC.

Comment (zhenggen-xu) [reply device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini:1]: OK, please test the HWSKU with QSFP DAC cable (not QSFPDD) for those 25G ports.   If results (port mapping and ports up status) good, we can close this comment.

Comment (byu343) [reply device/arista/x86_64-arista_7060dx5_32/Arista-7060DX5-32-25Gx96-100Gx8-200Gx8/port_config.ini:1]: Yes, this has been tested on QSFP DAC (100GBASE-CR4)

**Review by zhenggen-xu — APPROVED** [review]
